diff --git a/src/server-main.js b/src/server-main.js
index e3dd44e4123..c342659f93e 100644
--- a/src/server-main.js
+++ b/src/server-main.js
@@ -139,7 +139,12 @@ async function start() {
        });
        server.on('error', async (err) => {
                const remoteExtensionHostAgentServer = await getRemoteExtensionHostAgentServer();
-               return remoteExtensionHostAgentServer.handleServerError(err);
+               remoteExtensionHostAgentServer.handleServerError(err);
+                if (err.code === 'EADDRINUSE') {
+                        console.error('Terminating');
+                        process.exit(1);
+                }
+                return
        });

        const host = sanitizeStringArg(parsedArgs['host']) || (parsedArgs['compatibility'] !== '1.63' ? 'localhost' : undefined);
